server/lib/chat/chat-service.ts

```diff
@@ -321,6 +412,24 @@ export default class ChatService {
     user.unsubscribe(getChannelPath(channelName))
   }
 
+  private async removeUserFromChannel(
```

Comment:
As per above, this makes it so users can't be removed unless they're currently online. I think you should rewrite this to take a user ID and find the userSockets as part of the logic here (optionally, if they aren't found you just skip that part)